model/Check/System/AdvancedSearch/AbstractAdvancedSearchCheck.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoSystemStatus\model\Check\System\AdvancedSearch;
+
+use common_report_Report;
+use oat\oatbox\reporting\Report;
+use common_ext_ExtensionsManager;
+use Psr\Container\ContainerInterface;
+use oat\taoSystemStatus\model\Check\AbstractCheck;
+
+abstract class AbstractAdvancedSearchCheck extends AbstractCheck
```

Comment:
2 layers of abstraction?

Reply:
I don't want to repeat methods, like `isActive`, `getType`, `getCategory` - result must be the same for all checks